rest-api-model-ql/src/main/kotlin/org/modelix/sample/restapimodelql/Serialization.kt

```diff
@@ -54,14 +56,44 @@ fun List<N_Lecture>.toJson() = LectureList(this.mapNotNull {
     }
 })
 
+/**
+ * A data class to represent the type of change that was sent/received by the websocket
+ */
 enum class WhatChanged {
     ROOM,
     ROOM_LIST,
     LECTURE,
     LECTURE_LIST
 }
-
+/**
+ * A data class to represent the change notification sent/received by the update websocket
+ */
 data class ChangeNotification(
-    val whatChanged: WhatChanged,
-    val change: Any
+        val whatChanged: WhatChanged,
+        val change:  Any
 )
+/**
+ * A Gson deserializer which can distinguish between the different WhatChanged types.
+ * Will return the deserialized ChangeNotification
+ */
+class ChangeNotificationDeserializer() : JsonDeserializer<ChangeNotification> {
+
+    private var gson: Gson = Gson()
+    private var changeTypeRegistry: MutableMap<WhatChanged, Class<out Any?>> = HashMap()
+    private var changeTypeElementName: String = "whatChanged"
```

Comment:
```suggestion
    private val gson = Gson()
    private val changeTypeRegistry: MutableMap<WhatChanged, Class<out Any?>> = EnumMap(WhatChanged::class.java)
    private val changeTypeElementName = "whatChanged"
```
Variables can be val, removed inferable types, use EnumMap instead of HashMap for better performance.